mobius-core/src/main/java/com/spotify/mobius/FilteredEventSource.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius;
+
+import static com.spotify.mobius.internal_util.Preconditions.checkNotNull;
+
+import com.spotify.mobius.disposables.Disposable;
+import com.spotify.mobius.functions.Consumer;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * An internal class for wrapping another given event source, and handling subscribing/unsubscribing
+ * to it based on an active state. For use from {@link ConditionalFilteringEventSource}.
+ *
+ * @param <F> The Event class
+ */
+final class FilteredEventSource<F> implements EventSource<F> {
```

Comment:
How come this is a separate thing from ConditionalFilteringEventSource? Did it end up being very complex?

Reply:
Yes, it seemed to also mix responsibilities a little. Originally it was an internal non-static class in ConditionalFilteringEventSource, but when writing tests i realized how big it was getting, and that really I was testing two separate things at once - the wrapping around some decider-event-source + forwarding its events, and the actual managing of subscribe/unsubscribe from to-be-filtered event sources. Splitting it seemed to make sense, and tests became smaller and more readable.